src/main/ApplicationImpl.cpp

```diff
@@ -366,7 +349,7 @@ ApplicationImpl::~ApplicationImpl()
     reportCfgMetrics();
     shutdownMainIOContext();
     joinAllThreads();
-    if (!modeHasDatabase(mAppMode))
+    if (mAppMode == Application::AppMode::REPLAY_IN_MEMORY)
```

Comment:
this doesn't look right: we should not use app modes directly
So which flags really need to trigger this path?

NB: it may be that this doesn't actually belong here, and instead in the command line handling that actually does "fast replay mode"

Reply:
I don't think it's right to use flags here either: this codepath isn't about what kind of database we use, and whether we commit ledger txn or not. It is specific to "fast replay mode", which chooses to cleanup the buckets folder in the end. I agree that perhaps it just doesn't belong here, and should be moved to the command line code.

Reply:
Update: moving it to command line doesn't seem ideal either, as we need the app destructor to execute before cleaning up `buckets`, so we'd need to spin up the application again just to delete the folder...